src/glum/_solvers.py

```diff
@@ -151,9 +182,25 @@ def update_hessian(state, data, active_set):
             np.arange(data.X.shape[0], dtype=np.int32),
             active_set,
         )
-        state.hessian[np.ix_(active_set, active_set)] = hessian_init
+
+        # In the sparse Hessian case, state.hessian is a coo_matrix.
+        # hessian_init is a numpy array of size active_set x active_set;
+        # we can convert this to a coo_matrix and simply add it to the state.hessian
+        if state.use_sparse_hessian:
+            coo_data = hessian_init.flatten()
```

Comment:
Does it make sense to explicitly pass the flattening order here? I worry that if default behavior under the hood were to change then your repeat / tiling operations would yield invalid indices.